haystack/document_stores/weaviate.py

```diff
@@ -391,13 +392,15 @@ def _sanitize_id(self, id: str, index: Optional[str] = None) -> str:
 
     def _get_current_properties(self, index: Optional[str] = None) -> List[str]:
         """
-        Get all the existing properties in the schema.
+        Get all the existing properties in the schema, excluding those of type cross-reference
         """
         index = self._sanitize_index_name(index) or self.index
         cur_properties = []
         for class_item in self.weaviate_client.schema.get()["classes"]:
             if class_item["class"] == index:
-                cur_properties = [item["name"] for item in class_item["properties"]]
+                cur_properties = [
+                    item["name"] for item in class_item["properties"] if item["dataType"][0] in VALID_PROPERTY_TYPES
```

Comment:
Looks good, is dataType always guaranteed to be there and be a non-empty list? That's all that pops up at me here.